### PR TITLE
[3.9] Fix possible parent loop issues user groups

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #23933 cc @rolandalsace

### Summary of Changes

With this changes we make sure you can not configure an child group as your parent. As well as now use the namespaced classes.

### Testing Instructions

- Install Joomla using testing sample data (or create a few nested groups)
- check that the groups hierachie looks like this:
image
- now open the author group and check the allowed parent dropdown

### Expected result

After applying the patch anything this are children of the current group is going to be removed from that list.
image

### Actual result

Only the current group itself is removed from the list any other child is still there.
image

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by ladyjer** [comment]
I have tested this item :white_check_mark: successfully on 9d6cdcffb1c44cf3e6cbc36bfdf39f56cc8d6806<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24021">issues.joomla.org/tracker/joomla-cms/24021</a>.</sub>

**Comment by Quy** [comment]
It doesn't work for this case:
group1

group2

**Comment by zero-24** [comment]
@Quy @ladyjer please retest with the latest code provided. Please also test creation a new group

**Comment by Quy** [comment]
`Call to undefined method JFormFieldGroupParent::checkTheChildren()`

**Comment by zero-24** [comment]
You should not rename the method name after testing ... Fixed thanks @Quy

File: administrator/components/com_users/models/fields/groupparent.php [hunk 1]
```diff
@@ -3,13 +3,18 @@
  * @package     Joomla.Administrator
  * @subpackage  com_users
  *
- * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
```
Comment (HLeithner) [inline administrator/components/com_users/models/fields/groupparent.php]: moved back in time?

Comment (zero-24) [reply administrator/components/com_users/models/fields/groupparent.php]: Yes just for you :D

File: administrator/components/com_users/models/fields/groupparent.php [hunk 2]
```diff
@@ -26,6 +31,31 @@ class JFormFieldGroupParent extends JFormFieldList
 	 */
 	protected $type = 'GroupParent';
 
+	/**
+	 * Method to clean the Usergroup Options from all children starting by an given father
```
Comment (Quy) [inline administrator/components/com_users/models/fields/groupparent.php]:
```suggestion
	 * Method to clean the Usergroup Options from all children starting by a given father
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4494375e84c3bbae10de2243cd6c5d6dacad6d4e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24021">issues.joomla.org/tracker/joomla-cms/24021</a>.</sub>

**Comment by HLeithner** [comment]
@ladyjer could you please test this PR again?

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 38466d2d29bb8956153011cea6c0bed10614c504<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24021">issues.joomla.org/tracker/joomla-cms/24021</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by HLeithner** [comment]
@franz-wohlkoenig Quy has already tested this commit thx.

@zero-24 thx for this PR.

**Pull request merged**